src/helpers/workspace.ts

```diff
@@ -17,6 +25,49 @@ export function getWorkspaceRoot(ignoreException: boolean = false): string | und
   return workspaceRoot;
 }
 
+export async function getWorkspaces(): Promise<TruffleWorkspace[]> {
+  const workspaces: TruffleWorkspace[] = [];
+
+  await Promise.all(
+    workspace.workspaceFolders!.map(async (ws) => {
+      workspaces.push(...(await getWorkspaceFiles(ws.uri.fsPath)));
+    })
+  );
+
+  if (workspaces === undefined) {
+    const error = new Error(Constants.errorMessageStrings.VariableShouldBeDefined("Workspace root"));
+    Telemetry.sendException(error);
+    throw error;
+  }
+
+  return workspaces;
+}
+
 export function isWorkspaceOpen(): boolean {
   return !!(workspace.workspaceFolders && workspace.workspaceFolders[0].uri.fsPath);
 }
+
+async function getWorkspaceFiles(dirPath: string, truffleWorkSpaces?: TruffleWorkspace[]): Promise<TruffleWorkspace[]> {
+  const files = fs.readdirSync(dirPath);
+
+  truffleWorkSpaces = truffleWorkSpaces || [];
```

Comment:
should you mutate the variable like this internal to a function?

Reply:
I've realized Glob is already installed in the project. So, I revamped all this function and now it's more clear and with no mutations...